app/com/arpnetworking/metrics/portal/reports/impl/DatabaseReportRepository.java

```diff
@@ -176,6 +176,47 @@ public void close() {
         return getBeanReport(identifier, organization).map(models.ebean.Report::toInternal);
     }
 
+    @Override
+    public int deleteReport(final UUID identifier, final Organization organization) {
+        assertIsOpen();
+
+        LOGGER.debug()
+                .setMessage("Deleting report")
+                .addData("uuid", identifier)
+                .addData("organization.uuid", organization.getId())
+                .log();
+
+        final Optional<models.ebean.Report> report;
+        try (Transaction transaction = _ebeanServer.beginTransaction()) {
+            // Ebean does not generate the soft-delete correctly when using the query builder, instead
+            // we must fetch the report entity first and then call delete with that.
+            report = getBeanReport(identifier, organization);
+            report.ifPresent(r -> {
+                _ebeanServer.delete(r);
+                LOGGER.debug()
+                        .setMessage("Deleted report")
+                        .addData("uuid", identifier)
+                        .addData("organization.uuid", organization.getId())
+                        .log();
+            });
+            transaction.commit();
+            // CHECKSTYLE.OFF: IllegalCatchCheck
+        } catch (final RuntimeException e) {
+            // CHECKSTYLE.ON: IllegalCatchCheck
+            LOGGER.error()
+                    .setMessage("Failed to delete report")
+                    .addData("uuid", identifier)
+                    .addData("organization.uuid", organization.getId())
+                    .setThrowable(e)
+                    .log();
+            throw new PersistenceException(e);
```

Comment:
Is there a reason we wrap the exception instead of rethrowing? (not sure what the common practice in this project is at the moment)

Also, as a general rule of thumb you should either log or throw, but not both. If you throw, the caller can decide the severity and log (or not if they act) and your log would just confuse the matter. If you swallow the exception (e.g. recover, default, etc.), you need to log something at an appropriate level to indicate what happened.

I bet that Play will log the uncaught exception from the handler.

Reply:
^ Either that or remove the throw and return 0 (although I'm less of a fan of that approach)